tellerlogger/src/main/java/com/kiliansteenman/teller/logger/ui/TellerLogNotification.kt

```diff
@@ -96,13 +104,27 @@ class TellerLogNotification(
         context,
         NOTIFICATION_ID,
         TellerLogIntentFactory.createIntent(context),
-        getIntentFlags(),
+        PendingIntent.FLAG_UPDATE_CURRENT or immutableFlag(),
     )
 
-    private fun getIntentFlags() =
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
-        } else {
-            PendingIntent.FLAG_UPDATE_CURRENT
-        }
+    private fun createClearAction(): NotificationCompat.Action {
+        val clearLogBroadcastIntent = Intent(context, ClearLogActionBroadcastReceiver::class.java)
+        val pendingBroadcastIntent = PendingIntent.getBroadcast(
+            context,
+            CLEAR_INTENT_REQUEST_CODE,
+            clearLogBroadcastIntent,
+            PendingIntent.FLAG_ONE_SHOT or immutableFlag()
+        )
+        return NotificationCompat.Action(
+            R.drawable.ic_delete_white,
+            context.getString(R.string.notification_action_clear),
+            pendingBroadcastIntent
+        )
+    }
+
+    private fun immutableFlag() = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
Might be nicer to write:
```
private val immutableFlag
  get() = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) PendingIntent.FLAG_IMMUTABLE else 0
```

Reply:
looks nicer to me as well 😊  Made the changes 👍